python/pyspark/sql/pandas/serializers.py

```diff
@@ -190,7 +190,7 @@ def arrow_to_pandas(self, arrow_column, struct_in_pandas="dict", ndarray_as_list
         )
         return converter(s)
 
-    def _create_array(self, series, arrow_type, spark_type=None):
+    def _create_array(self, series, arrow_type, spark_type=None, arrow_cast=False):
```

Comment:
I'm wondering if this needs to take the argument as it is a method and `self` should have `_arrow_cast`.

Reply:
Makes sense, refactored.

Reply:
On second thought, shall we revert the last commit [9d359b9](https://github.com/apache/spark/pull/41503/commits/9d359b99d2dd53081da9aaa803c28634ed3ddab0)?
I feel weird that `arrow_to_pandas` takes parameters `struct_in_pandas` and `ndarray_as_list` to control its behavior, whereas `_create_array` uses `self._arrow_cast`.

Reply:
Sounds good! Per offline discussion, I'll rever the last commit and refactor it later.